src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -105,7 +105,7 @@ def command_callable_decorator(*args, **kw):
                     result = command_callable(*args, **kw)
                     self.process_result(result)
                     execution.complete(ExecutionStatus.SUCCESS)
-                except Exception as err:
+                except BaseException as err:
```

Comment:
Is it bulletproof? We were thinking about registering exit handlers.

Is it possible to add tests?

Reply:
can always be forcefully killed - but in that case, we wouldn't get metrics at all.
I will add a test for this.

Reply:
Added unit tests